### PR TITLE
fix(ui): improve JSON copy functionality in state explorer - AB-242

### PR DESCRIPTION
**Pull request by VladKrasny**
## Summary by CodeRabbit

- New Features
  - JSON Viewer: copy selected portions or the entire JSON to clipboard; output is pretty-printed and selection-aware.

- Improvements
  - Stricter plain-object detection across the UI and file-encoding checks to reduce edge-case behavior.
  - Selection-to-JSON extraction for precise subset copying.

- Refactor
  - Centralized plain-object check utility for consistent behavior.

- Tests
  - Added comprehensive tests covering selection-based copying scenarios and edge cases.

### CODE REVIEW

File: src/ui/src/components/shared/SharedJsonViewer/SharedJsonViewer.vue [hunk 1]
```diff
@@ -99,8 +103,33 @@ const isRoot = computed(() => props.path.length === 0 && !props.hideRoot);
...
+) {
+	const selection = window.getSelection();
+	const selectionText = selection?.toString().trim() ?? "";
+
+	const json = selectionText
+		? selectionToJson(arrayOrObject, selectionText)
+		: arrayOrObject;
+
+	copy(prettyJson(json));
+}
```
Comment (madeindjs) [inline src/ui/src/components/shared/SharedJsonViewer/SharedJsonViewer.vue]: I think we should avoid copying empty objects when I selecting a simple value:

image

So you probably need to avoid the `.prevent` modifier and prevent the default event only if `prettyJson(json) !== '{}'`

File: src/ui/src/components/shared/SharedJsonViewer/SharedJsonViewer.vue [hunk 2]
```diff
@@ -99,8 +104,49 @@
...
+	const subset = selectionToJson(arrayOrObject, selectionText);
+
+	if (isEmptySubset(subset)) {
+		return; // nothing meaningful recognized → allow native copy of raw text
+	}
+
+	event.preventDefault();
+
+	copy(prettyJson(subset)); // copy subset
+}
```
[the remark this answers is not in the document]
Comment (VladKrasny) [reply src/ui/src/components/shared/SharedJsonViewer/SharedJsonViewer.vue:151]: vueuse already does copy logic, so I don't see any reason to implement it above and use the lib as a fallback.
The same goes for the suggestion to copy the full JSON. We should only handle selection copying and not intercept the copy event when nothing is selected.
